Clarify build instructions (#1530)
opengisch/QField: README.md

### For Android

Use the dockerized QField SDK.

```
QFIELD_SDK=$(awk -F "=" '/osgeo4a_version/{print $2}' sdk.conf)
ANDROID_NDK_PLATFORM=android-29

git submodule update --init --recursive

docker run -e ANDROID_NDK_PLATFORM -v $(pwd):/usr/src/qfield docker.io/opengisch/qfield-sdk:$QFIELD_SDK /usr/src/qfield/scripts/docker-build.sh --rm
```

This will put the apk into a subfolder `build-docker/out/build/outputs/apk`

### For Desktop

To build QField for a desktop environment:

* Get QGIS development libraries. QField normally uses
  bleeding edge QGIS code. Ideally install nightly builds

>   or compile it on your own and install to a custom prefix path.

Reviewer: ```suggestion
  or compile it on your own and install to a custom prefix path. See more about QGIS compilation [here](https://github.com/qgis/QGIS/blob/master/INSTALL.md).
```

> 

Reviewer: ```suggestion
* install QField dependencies:

```bash
$ sudo apt install qtquickcontrols2-5-dev libqt5sensors5-dev qml-module-qtpositioning
```

```

Author: Can we separate that away since its platform specific?

Reviewer: Sure! 

Since QField requires Qt 5.14, at least on Ubuntu LTS the dev needs to download Qt libs from the their website (https://www.qt.io/download) and all the requirements are already bundled there.

The list above is relevant when using the system's Qt version.

Worth mentioning this somehow in the text.

Author: That sounds like information that is temporary and needs maintenance?

Shouldn't we just document the system approach?

Author: What I was referring to before is that on my fedora system things are different.

Reviewer: Now I see your point.
Can we at least describe the approach where qt.io download is used since everything is shipped there, is not platform specific and will rarely change?

Author: If we can do it this way, yes 👍

> * Download the most recent Qt 5.x version from the [Qt Website](https://www.qt.io/download) to satisfy QField dependencies. Setup a new QtCreator kit that uses the downloaded Qt libs.

Author: @suricactus two questions:

1. does this work reliably? Don't you need to recompile other dependencies too (like QScintilla, QWT, QtWebKit) in order to get QGIS to compile?
2. on my system I don't need to do this because Qt is new enough, should we flag this instruction as optional?

Reviewer: 1) for me it was plug-n-play on two Ubuntu machines. At some point I had QField, QGIS and the rest of the Qt dependencies coming compiled with three different versions and it worked. We had this discussion a while ago, as far as  I remember @signedav also confirmed this with Qt 5.13+.
2) sure! I wish I can use my system qt libs for development too. Ubuntu LTS is already lacking 3 versions behind.

> * Download the most recent Qt 5.x version from the [Qt Website](https://www.qt.io/download) to satisfy QField dependencies. Setup a new QtCreator kit that uses the downloaded Qt libs.

Author: ```suggestion
* If your system comes with a Qt version which is too low for QField, you can also install Qt manually. Download the most recent Qt 5.* version from the [Qt Website](https://www.qt.io/download) to satisfy QField dependencies. Setup a new QtCreator kit that uses the downloaded Qt libs.
```

```sh
cd QField
git submodule update --init --recursive
```

>  * Open CMakeList.txt with QtCreator

Reviewer: ```suggestion
 * Open `CMakeList.txt` with QtCreator
```

 * Hit build

> If you make your own QGIS build, use the following variables: `QGIS_PREREFIX_PATH`.

Reviewer: ```suggestion
If you make your own QGIS build, use the following variables: `QGIS_PREFIX_PATH`.
```
